### ISSUE TITLE
How does Godot 3.0  write 2D shader

### ISSUE THREAD

**Issue by cjmxp** [godotengine/godot#10073]
**Operating system or device - Godot version:**

How does Godot 3.0  write 2D shader? Where does the new shader syntax get help documentation?

**Comment by eon-s** [comment]
No documentation yet (as explained in the post about the alpha) but you can look here:
https://github.com/godotengine/godot/blob/772485cdb3a0b0003da6e174157680559fa2a59e/servers/visual/shader_types.cpp#L153

And maybe here too: 
https://github.com/godotengine/godot/blob/e577c5b0705168177943fcdf9a0b66c1f8f864f3/drivers/gles3/shader_compiler_gles3.cpp

The error messages are better than on 2.x and has code completion too.

Basically, the shaders are now like:
```
shader_type canvas_item;

void vertex() {
}

void fragment() {
}

void light() {
}
```

in the same place.
You can find some examples in the demos repository (I think "screen space shaders" is working).

**Comment by curly-brace** [comment]
i've created a reference based on the sources (there are 3 sheets in doc)
https://docs.google.com/spreadsheets/d/1hBpkNor5-WUTKuiEP3Myo4wzleCWvqPqn1OLgOwFsmU/edit#gid=1226435562
